core/src/reporter/reporter.ts

```diff
@@ -61,6 +68,20 @@ export function reporter(state: State, logger: Logger) {
         }
       }
     }
+
```

Comment:
I believe this reporter class is also used for all services. It seems this will through error in case of VRF and RR. Did you consider this case or I'm missing your point?